### PR TITLE
Support IPv6StaticDefaultGateways

### PR DESCRIPTION
**Pull request by sunharis**
Currently there is no support to setting up the Static Default IPv6 gateway via redfish.

This commit adds IPv6StaticDefaultGateways parameter to the ethernet interface, on which user can send PATCH request and setup the Static IPv6 gateway for the interface.

Backend does not support the DefaultGateway6 as array. So this commit ignores the PrefixLength and just sets the Address on to the dbus property and corresponding network file.

Tested by:
 PATCH https://${bmc}/redfish/v1/Managers/bmc/EthernetInterfaces/<id>
 -d '{"IPv6StaticDefaultGateways": [{"Address": "2002:903:15F:325:9:3:29:1",
 "PrefixLength": 24}]}'

 PATCH https://${bmc}/redfish/v1/Managers/bmc/EthernetInterfaces/<id>
 -d '{"IPv6StaticDefaultGateways": [{"Address": "2002:903:15F:325:9:3:29:1"}]}'

 PATCH https://${bmc}/redfish/v1/Managers/bmc/EthernetInterfaces/<id>
 -d '{"IPv6StaticDefaultGateways": [{"PrefixLength": 24}]}'
 --> this will return PropertyMissing error

 GET https://${bmc}/redfish/v1/Managers/bmc/EthernetInterfaces/<id>
 --> returns IPv6 static gateway

 Redfish validator passed.

Change-Id: If6aaa6981a9272a733594f0ee313873a09f67758

### CODE REVIEW

**Comment by sunharis** [comment]
Upstream commit: https://gerrit.openbmc.org/c/openbmc/bmcweb/+/62255

**Review by raviteja-b — APPROVED** [review]

File: redfish-core/lib/ethernet.hpp [hunk 1]
```diff
@@ -840,6 +841,81 @@ inline void createIPv6(const std::string& ifaceId, uint8_t prefixLength,
...
+    BMCWEB_LOG_DEBUG << "IPv6 Static Gateway to set: " << *address;
+
+    // TODO: Backend currently supports single DefaultGateway6 as a string.
+    // This needs a change when backend implements this as an array
+    // of multiple IPv6 default gateway addresses.
+    crow::connections::systemBus->async_method_call(
+        [asyncResp](const boost::system::error_code ec) {
+        if (ec)
+        {
+            messages::internalError(asyncResp->res);
```
Comment (gtmills) [inline redfish-core/lib/ethernet.hpp:907]: Would a trace help debug 
        BMCWEB_LOG_ERROR << "D-Bus Set DefaultGateway" << ec.msg;

File: redfish-core/lib/ethernet.hpp [hunk 2]
```diff
@@ -840,6 +841,81 @@ inline void createIPv6(const std::string& ifaceId, uint8_t prefixLength,
...
+            return;
+        }
+        asyncResp->res.result(boost::beast::http::status::no_content);
+        },
+        "xyz.openbmc_project.Network",
+        "/xyz/openbmc_project/network/" + ifaceId,
+        "org.freedesktop.DBus.Properties", "Set",
+        "xyz.openbmc_project.Network.EthernetInterface", "DefaultGateway6",
+        dbus::utility::DbusVariantType(*address));
```
Comment (gtmills) [inline redfish-core/lib/ethernet.hpp:916]: If it is null or empty, don't you just fall through the if statement above and try to set this?

Comment (gtmills) [reply redfish-core/lib/ethernet.hpp]: Moved to inside the if

File: redfish-core/lib/ethernet.hpp [hunk 3]
```diff
@@ -840,6 +841,81 @@ inline void createIPv6(const std::string& ifaceId, uint8_t prefixLength,
...
+                    pathString);
+                return;
+            }
+
+            if (!address)
+            {
+                messages::propertyMissing(asyncResp->res,
+                                          pathString + "/Address");
+                return;
+            }
```
[the remark this answers is not in the document]
Comment (gtmills) [reply redfish-core/lib/ethernet.hpp:895]: moved to inside the if

**Review by gtmills — APPROVED** [review]

File: redfish-core/lib/ethernet.hpp [hunk 4]
```diff
@@ -840,6 +841,82 @@ inline void createIPv6(const std::string& ifaceId, uint8_t prefixLength,
...
+ * @return None
+ */
+
+inline void handleIPv6DefaultGateway(
+    const std::string& ifaceId, const nlohmann::json& input,
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+{
+    if (!input.is_array() || input.empty())
+    {
+        BMCWEB_LOG_ERROR << "Not array";
```
Comment (gtmills) [inline redfish-core/lib/ethernet.hpp:860]: upstream this should be warning, might be good to that here too

**Pull request merged**